lib/client.js

```diff
@@ -88,9 +91,86 @@ function defaultId () {
   return 'mqttjs_' + Math.random().toString(16).substr(2, 8)
 }
 
+function applyTopicAlias (client, packet) {
+  if (client.options.protocolVersion === 5) {
+    if (packet.cmd === 'publish') {
+      var alias
+      if (packet.properties) {
+        alias = packet.properties.topicAlias
+      }
+      var topic = packet.topic.toString()
+      if (client.topicAliasSend) {
+        if (alias) {
+          if (topic.length !== 0) {
+            // register topic alias
+            debug('applyTopicAlias :: register topic: %s - alias: %d', topic, alias)
+            if (!client.topicAliasSend.put(topic, alias)) {
+              debug('applyTopicAlias :: error out of range. topic: %s - alias: %d', topic, alias)
+              return new Error('Sending Topic Alias out of range')
+            }
+          }
+        } else {
+          if (topic.length !== 0) {
+            if (client.options.autoAssignTopicAlias) {
+              alias = client.topicAliasSend.getAliasByTopic(topic)
+              if (alias) {
+                packet.topic = ''
+                packet.properties = {...(packet.properties), topicAlias: alias}
+                debug('applyTopicAlias :: auto assign(use) topic: %s - alias: %d', topic, alias)
+              } else {
+                alias = client.topicAliasSend.getLruAlias()
+                client.topicAliasSend.put(topic, alias)
+                packet.properties = {...(packet.properties), topicAlias: alias}
+                debug('applyTopicAlias :: auto assign topic: %s - alias: %d', topic, alias)
+              }
+            } else if (client.options.autoUseTopicAlias) {
+              alias = client.topicAliasSend.getAliasByTopic(topic)
+              if (alias) {
+                packet.topic = ''
+                packet.properties = {...(packet.properties), topicAlias: alias}
+                debug('applyTopicAlias :: auto use topic: %s - alias: %d', topic, alias)
+              }
+            }
+          }
+        }
+      } else if (alias) {
+        debug('applyTopicAlias :: error out of range. topic: %s - alias: %d', topic, alias)
+        return new Error('Sending Topic Alias out of range')
+      }
+    }
+  }
+}
+
+function removeTopicAlias (client, packet) {
+  // remove topic alias because it shouldn't be used on re-sending
```

Comment:
can you add a bit of a better comment here. What's the use for this function? Is it removing a topic alias because some other topic has overridden the topic?

Reply:
This is a kind of utility function for internal use.
I think that the commend should be removed here. 
This function removes Topic Alias property and recover Topic Name from the Topic Alias.
Perhaps the function name should be `removeTopicAliasAndRecoverTopicName`. Is this too long?
I think that the comment should be written the caller side.

Reply:
done.